In [2]:
import pandas as pd

autos = pd.read_json("../Data/autos.json")

In [42]:
autos 

aspiration   body-style  bore  city-mpg  compression-ratio  curb-weight  \
0          std  convertible  3.47        21                9.0         2548   
1          std  convertible  3.47        21                9.0         2548   
2          std    hatchback  2.68        19                9.0         2823   
3          std        sedan  3.19        24               10.0         2337   
4          std        sedan  3.19        18                8.0         2824   
..         ...          ...   ...       ...                ...          ...   
200        std        sedan  3.78        23                9.5         2952   
201      turbo        sedan  3.78        19                8.7         3049   
202        std        sedan  3.58        18                8.8         3012   
203      turbo        sedan  3.01        26               23.0         3217   
204      turbo        sedan  3.78        19                9.5         3062   

    drive-wheels engine-location  engine-size engine-type  ...         make  \
0            rwd           front          130        dohc  ...  alfa-romero   
1            rwd           front          130        dohc  ...  alfa-romero   
2            rwd           front          152        ohcv  ...  alfa-romero   
3            fwd           front          109         ohc  ...         audi   
4            4wd           front          136         ohc  ...         audi   
..           ...             ...          ...         ...  ...          ...   
200          rwd           front          141         ohc  ...        volvo   
201          rwd           front          141         ohc  ...        volvo   
202          rwd           front          173        ohcv  ...        volvo   
203          rwd           front          145         ohc  ...        volvo   
204          rwd           front          141         ohc  ...        volvo   

    normalized-losses  num-of-cylinders  num-of-doors  peak-rpm    price  \
0                 NaN              four           two    5000.0  13495.0   
1                 NaN              four           two    5000.0  16500.0   
2                 NaN               six           two    5000.0  16500.0   
3               164.0              four          four    5500.0  13950.0   
4               164.0              five          four    5500.0  17450.0   
..                ...               ...           ...       ...      ...   
200              95.0              four          four    5400.0  16845.0   
201              95.0              four          four    5300.0  19045.0   
202              95.0               six          four    5500.0  21485.0   
203              95.0               six          four    4800.0  22470.0   
204              95.0              four          four    5400.0  22625.0   

    stroke  symboling wheel-base width  
0     2.68          3       88.6  64.1  
1     2.68          3       88.6  64.1  
2     3.47          1       94.5  65.5  
3     3.40          2       99.8  66.2  
4     3.40          2       99.4  66.4  
..     ...        ...        ...   ...  
200   3.15         -1      109.1  68.9  
201   3.15         -1      109.1  68.8  
202   2.87         -1      109.1  68.9  
203   3.40         -1      109.1  68.9  
204   3.15         -1      109.1  68.9  

[205 rows x 26 columns]

### Q1: How many cars from Jaguar are in the dataset, and what is the price of the most expensive one?

### Answer
###   count=3    price=36000.0

In [28]:
jaguar_count = autos.query("`make` == 'jaguar'").count()
print(f"count={jaguar_count['make']}")

jaguar_max_price = autos.query("`make` == 'jaguar'")["price"].max()
print(f"price={jaguar_max_price}")

count=3
price=36000.0


### Q2: How many cars from Toyota are in the dataset, and what is the price of the most expensive one?

### Answer
###   count=32    price=17669.0

In [25]:
toyota_count = autos.query("`make` == 'toyota'").count()
print(f"count={toyota_count['make']}")

toyota_max_price = autos.query("`make` == 'toyota'")["price"].max()
print(f"price={toyota_max_price}")



count=32
price=17669.0


### Q3: What is the length, width and height of the most expensive car in the entire dataset?

### Answer
###   length=199.2    width=72.0    height=55.4

In [56]:
max_price = autos.loc[autos["price"].idxmax()]
car_length = max_price["length"]
car_width = max_price["width"]
car_height = max_price["height"]
print(f"length={car_length} width={car_width} height={car_height}")

length=199.2 width=72.0 height=55.4


### Q4: What is the lowest price per horsepower in the dataset, and what brand ("make") is that car?

### Answer
###   price per horsepower=72.84    brand=Toyota

In [94]:
autos['price_per_hp'] = autos['price'] / autos['horsepower']
lowest_price_hp = autos['price_per_hp'].min()
price_per_hp = autos['price_per_hp'].idxmin()
hp_brand = autos.loc[price_per_hp, 'make']

print(f"price per horsepower={lowest_price_hp} brand={hp_brand}")


price per horsepower=72.83620689655173 brand=toyota


### Q5: How many of the cars in the dataset has as many cylinders as they have doors?

### Answer
###   cars=95

In [96]:
matching_parts = autos[autos["num-of-cylinders"] == autos["num-of-doors"]]
num_of_matching_cars = matching_parts.shape[0]

print(f"cars={num_of_matching_cars}")

cars=95


### T1: Print a string to inform the user of the price difference between the cheapest and the most expensive car in the dataset.

### Solution
###   Example:
###   The price difference between the cheapest and the most expensive car is 40282.0

In [100]:
max_car_price = autos["price"].max()
min_car_price = autos["price"].min()
difference_price = max_car_price - min_car_price

print(f"the difference between the cheapest and the most expensive car is {difference_price}")

the difference between the cheapest and the most expensive car is 40282.0


### T2: Ask the user to input a brand, then print the price range for that brand.

Solution
   Example 1:
   Input the name of a brand: volvo
   The prices for cars of brand 'volvo' ranges from 12940.0 to 22625.0

   Example 2:
   Input the name of a brand: toyota
   The prices for cars of brand 'toyota' ranges from 5348.0 to 17669.0

   Example 3:
   Input the name of a brand: tesla
   The brand 'tesla' does not exists in the dataset.

In [106]:
user_input_brand = input("Input the name of a brand:")

selected_brand = autos[autos['make'] == user_input_brand] # tagen av gpt, måste lära dig mer!

if not selected_brand.empty:
    min_brand_price = selected_brand['price'].min()
    max_brand_price = selected_brand['price'].max()

    print(f"the prices for cars of brand {user_input_brand} ranges from {min_brand_price} to {max_brand_price}")

else:
    print(f"The brand {user_input_brand} does not exist in the dataset")



The brand bentley does not exist in the dataset


### T3: Ask the user to input a brand, then print the number of cars in the dataset for that brand, and all attributes for a random sample car of that brand.

Solution
   Example:
   Input the name of a brand: mazda
   There are 17 cars of brand 'mazda' in the dataset.

   Here is the data for a random 'mazda' car:
   - aspiration = std
   - body-style = sedan
   - bore = 3.03
   - city-mpg = 31
   - compression-ratio = 9.0
   - curb-weight = 1945
   - drive-wheels = fwd
   - engine-location = front
   - engine-size = 91
   - engine-type = ohc
   - fuel-system = 2bbl
   - fuel-type = gas
   - height = 54.1
   - highway-mpg = 38
   - horsepower = 68.0
   - length = 166.8
   - make = mazda
   - normalized-losses = 113.0
   - num-of-cylinders = four
   - num-of-doors = four
   - peak-rpm = 5000.0
   - price = 6695.0
   - stroke = 3.15
   - symboling = 1
   - wheel-base = 93.1
   - width = 64.2

In [ ]:
import random



In [ ]:
numbers_df = pd.DataFrame(np.random.randint(low=1, high=100, size=[5, 5]))
numbers_df.loc[[0, 3], [1, 4]] =a np.nan
numbers_df